bin/check-zookeeper-cluster.rb

```diff
@@ -0,0 +1,152 @@
+#
+# DESCRIPTION:
+#   Check if a exhibitor managed Zookeeper cluster is OK.
+#
+#   This check will get exhibitor status information,
+#   and check each seperate node for ruok.
+#   This check also will compare cluster node count with a spesific value,
+#   avg latency from mntr for a threshold value, and If the cluster
+#   has a leader.
+#
+# PLATFORMS:
+#   All
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#  Check if a node has Zookeeper running and responds with imok.
+#
+#  Cluster should have 3 nodes, exhibitor status endpoint is not default, and average latency threshold should be 10
+#  ./check-zookeeeper-cluster.rb -c 3 -e http://localhost:8181/exhibitor/v1/cluster/status -l 10
+#
+
+require 'sensu-plugin/check/cli'
+require 'socket'
+require 'net/http'
+require 'json'
+
+class CheckZookeeperCluster < Sensu::Plugin::Check::CLI
+  option :count,
+         description: 'Zookeeper cluster node count',
+         short: '-c count',
+         long: '--count count',
+         default: 2181,
+         proc: proc(&:to_i)
+
+  option :exhibitor,
+         description: 'exhibitor end node for status checks',
+         short: '-e Exhibitor status end point',
+         long: '--exhibitor status end point',
+         default: 'http://localhost/exhibitor/v1/cluster/status'
+
+  option :latency,
+         description: 'Critical threshold for Zookeeper average latency',
+         short: '-l TICKS',
+         long: '--latency TICKS',
+         proc: proc(&:to_i),
+         default: 10
+
+  option :timeout,
+         description: 'How long to wait for a reply in seconds.',
+         short: '-t SECS',
+         long: '--timeout SECS',
+         proc: proc(&:to_i),
+         default: 5
+
+  def zookeeper_latency(server, port)
+    l = 0
+    TCPSocket.open(server, port) do |socket|
+      socket.write 'mntr'
+      ready = IO.select([socket], nil, nil, config[:timeout])
+      if ready.nil?
+        critical %(Zookeeper did not respond to 'mntr' within #{config[:timeout]} seconds)
+      end
+      l = ready.first.first.read.chomp.split("\n")[1].split("\t")[1].to_i
+    end
+  end
+
+  def check_ruok(server, port)
+    result = false
+    TCPSocket.open(server, port) do |socket|
+      socket.write 'ruok'
+      ready = IO.select([socket], nil, nil, config[:timeout])
+
+      if ready.nil?
+        critical %(Zookeeper did not respond to 'ruok' within #{config[:timeout]} seconds)
+      end
+
+      result = ready.first.first.read.chomp
+    end
+    result == 'imok'
+  end
+
+  def _leader_count(json)
+    l_count = max_latency = 0
+    json.each do |zk|
+      l_count += 1 if zk['isLeader']
+      l = zookeeper_latency(zk['hostname'], 2181)
+      max_latency = [max_latency, l].max
+    end
+    [l_count, max_latency]
+  end
+
+  def _check_leader(json)
+    e = []
+    l_count, max_latency = _leader_count(json)
+    l_count != 1 &&
+      e.push("cluster should have a leader (#{l_count})")
+    max_latency > config[:latency] &&
```

Comment:
These strangeness is all because rubocop did not like single line ifs and long lines at the same time. I think that was my bad running rubocop in another scope. Changing these to ifs.